src/pages/nft/class/_classId/index.vue

```diff
@@ -1084,7 +1088,9 @@ export default {
     },
     async handleCollectFromEdition(selectedValue, giftInfo = undefined) {
       const editions = this.getNFTBookStorePricesByClassId(this.classId) || {};
-      const edition = editions[selectedValue ?? this.selectedValue];
+      const index = selectedValue ?? this.selectedValue;
+      const edition =
+        editions.find(e => e.index === Number(index)) || editions[index];
```

Comment:
Can be refactored?

Reply:
Actually I'm not sure if selectedValue must be number, if yes this can be simpler